lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if info =~ ~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/ do
```

Comment:
We can drop `\|\|\|\|\|\|\|` marker, it's optional and always present together with other markers, not on its own.